paymentsheet/src/main/java/com/stripe/android/paymentsheet/LinkHandler.kt

```diff
@@ -0,0 +1,264 @@
+package com.stripe.android.paymentsheet
+
+import androidx.activity.result.ActivityResultCaller
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.SavedStateHandle
+import com.stripe.android.link.LinkActivityResult
+import com.stripe.android.link.LinkPaymentDetails
+import com.stripe.android.link.LinkPaymentLauncher
+import com.stripe.android.link.model.AccountStatus
+import com.stripe.android.link.ui.inline.UserInput
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.payments.paymentlauncher.PaymentResult
+import com.stripe.android.paymentsheet.analytics.EventReporter
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.state.LinkState
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel.Companion.SAVE_PROCESSING
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel.Companion.SAVE_SELECTION
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+internal class LinkHandler @Inject constructor(
+    val linkLauncher: LinkPaymentLauncher,
+    private val savedStateHandle: SavedStateHandle,
+    private val eventReporter: EventReporter,
+) {
+    sealed class ProcessingState {
+        object Ready : ProcessingState()
+
+        object Launched : ProcessingState()
+
+        object Started : ProcessingState()
+
+        class PaymentDetailsCollected(val details: LinkPaymentDetails.New?) : ProcessingState()
+
+        data class Error(val message: String?) : ProcessingState()
```

Comment:
Why do we need data class here

Reply:
It's internal, so no new public APIs. Doesn't have to be a data class, just made testing easier.